### ISSUE TITLE
Start process fail with `Process config loading failed`

### ISSUE THREAD

**Issue by Ridermansb** [Unitech/pm2#2964]
**What's going wrong?**
Pm2 on docker fail to load ecosystem.config.js

```
/app # pm2 start ecosystem.config.js
[PM2][WARN] Applications api, mqtt not running, starting...
[PM2] Process config loading failed []
[PM2] Process config loading failed []
┌──────────┬────┬──────┬─────┬────────┬─────────┬────────┬─────┬─────┬──────────┐
│ App name │ id │ mode │ pid │ status │ restart │ uptime │ cpu │ mem │ watching │
└──────────┴────┴──────┴─────┴────────┴─────────┴────────┴─────┴─────┴──────────┘
 Use `pm2 show <id|name>` to get more details about an app
```

**How could we reproduce this issue?**

Using this config ...

```
const nodeEnv = process.env.NODE_ENV || 'development';

if (nodeEnv === 'development') {
  require('dotenv').config();
}

const maxMemory = process.env.WEB_MEMORY || 80;

const base = {
  source_map_support: nodeEnv !== 'production',
  node_args: [
    '--optimize_for_size',
    '--max_old_space_size=400',
    '--gc_interval=100',
  ],
  instances: process.env.WEB_CONCURRENCY || -1,
  exec_mode: nodeEnv === 'production' ? 'cluster' : 'fork',
  max_memory_restart: `${maxMemory}M`,
};

module.exports = {
  apps: [
    Object.assign({}, {
      name: 'api',
      script: 'api/index.js',
      env: {
        NODE_ENV: nodeEnv,
        PORT: process.env.PORT || 3000,
        API_MONGO_URL: process.env.API_MONGO_URL,
        MAILGUN_API: process.env.MAILGUN_API,
        MAILGUN_DOMAIN: process.env.MAILGUN_DOMAIN,
        MAILGUN_FROM: process.env.MAILGUN_FROM,
        JWT_SECRET: process.env.JWT_SECRET,
        SALT_WORK_FACTOR: process.env.SALT_WORK_FACTOR,
      },
      env_production: {
        NODE_ENV: "production",
      },
    }, base),
    Object.assign({}, {
      name: 'mqtt',
      script: 'mqtt/mqtt.babel.js',
      env: {
        NODE_ENV: nodeEnv,
        MQTT_PORT: process.env.MQTT_PORT || 1886,
        MQTT_HOST: process.env.MQTT_HOST || `mqtt://localhost:${1886}`,
        MQTT_MONGO_URL: process.env.MQTT_MONGO_URL || 'mongodb://localhost/mqtt',
        MQTT_HTTP_PORT: process.env.MQTT_HTTP_PORT || 9001,
      },
      env_production: {
        NODE_ENV: "production",
      },
    }, base),
  ],
};
```

After build this docker ...

```
FROM keymetrics/pm2-docker-alpine:7

ARG NODE=production

ENV NODE_ENV $NODE
ENV MQTT_PORT 1886
ENV MQTT_HTTP_PORT 9001
ENV PORT 3000

## Update
RUN apk update && \
    apk upgrade && \
    apk add \
      git \
      bash

# Update pm2
RUN npm install pm2@2.5.0 -g

# Add DockerCompose
RUN apk add make py-pip && pip install docker-compose

COPY . ./

EXPOSE $MQTT_PORT $MQTT_HTTP_PORT $MQTT_HTTP_PORT

CMD pm2-docker ecosystem.config.js --env $NODE_ENV
```

When I enter into container and try manual start the process..

```
/app # pm2 start ecosystem.config.js
[PM2][WARN] Applications api, mqtt not running, starting...
[PM2] Process config loading failed []
[PM2] Process config loading failed []
┌──────────┬────┬──────┬─────┬────────┬─────────┬────────┬─────┬─────┬──────────┐
│ App name │ id │ mode │ pid │ status │ restart │ uptime │ cpu │ mem │ watching │
└──────────┴────┴──────┴─────┴────────┴─────────┴────────┴─────┴─────┴──────────┘
 Use `pm2 show <id|name>` to get more details about an app
```

## Supporting information

```
/app # uname -a
Linux 6320bf95d141 4.4.66-boot2docker #1 SMP Fri May 5 20:44:25 UTC 2017 x86_64 Linux
/app # node -v
v7.10.0
/app # pm2 -v
2.5.0
```

**Comment by vmarchaud** [comment]
Thanks for the detailed issue, is the issue happening without docker as well ?

**Comment by Ridermansb** [comment]
No, just when I use docker.
I found the problem.. i think is because of `instances` value (is -1). When I put "max" works.

**Comment by Unitech** [comment]
https://github.com/Unitech/pm2/issues/2953

**Comment by Unitech** [comment]
Published on PM2 2.6.0:

```bash
$ npm install pm2@latest -g
$ pm2 update
```
